pkg/controller/core/workload_controller.go

```diff
@@ -153,7 +156,14 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	}
 
 	if workload.SyncAdmittedCondition(&wl) {
-		return ctrl.Result{}, workload.ApplyAdmissionStatus(ctx, r.client, &wl, true)
+		if err := workload.ApplyAdmissionStatus(ctx, r.client, &wl, true); err != nil {
+			return ctrl.Result{}, err
+		}
+		if workload.IsAdmitted(&wl) {
+			waitTime := time.Since(wl.CreationTimestamp.Time)
```

Comment:
I just realized that this is wrong sometimes. The wait time is either the CreationTimestamp or the LastTransitionTime for the WorkloadEvicted, if it exists.

Reply:
The problem is that the evicted cond is updated when it gets the new reservation, let's add tome since reservation for"Admitted"